roles/openshift_etcd_migrate/templates/validatemigration.py.j2

```diff
@@ -0,0 +1,91 @@
+from subprocess import PIPE, Popen
+import logging
+import json
+import base64
+
+#logging.basicConfig(level=logging.INFO)
+
+def runCommand(cmd):
+	process = Popen(cmd, stderr=PIPE, stdout=PIPE, shell=True)
+	stdout, stderr = process.communicate()
+	rt = process.returncode
+
+	return stdout, stderr, rt
+
+def keyExists(etcddata):
+	if "count" not in etcddata:
+		return False
+	if etcddata["count"] < 1:
+		return False
+	if "kvs" not in etcddata:
+		return False
+	if len(etcddata["kvs"]) < 1:
+		return False
+	if "value" not in etcddata["kvs"][0]:
+		return False
+	if "key" not in etcddata["kvs"][0]:
+		return False
+
+	return True
+
+# TODO(jchaloup): once done ping kwoodson to test the etcd migration role
+
+if __name__ == "__main__":
+
+	ectl2="/usr/bin/etcdctl --cert-file {{ etcd_peer_cert_file }} --key-file {{ etcd_peer_key_file }} --ca-file {{ etcd_peer_ca_file }} -C https://{{ etcd_peer }}:2379"
+	ectl3="/usr/bin/etcdctl --cert {{ etcd_peer_cert_file }} --key {{ etcd_peer_key_file }} --cacert {{ etcd_peer_ca_file }} --endpoints 'https://{{ etcd_peer }}:2379'"
+
+	so, se, rc = runCommand("%s ls -r | sort" % ectl2)
+	for line in so.split("\n"):
+		if line == "":
+			continue
+
+		v2key = line
+
+		so, se, rc = runCommand("%s get %s 2>/dev/null" % (ectl2, v2key))
+		# check if the key is a directory
+		# if a rc of etcdctl get is 1 => the key is a directory
+		# if a rc of etcdctl get is 4 => the key does not exist
+		if rc == 1:
+			logging.info("Skipping directory %s" % v2key)
+			continue
+
+		if rc != 0:
+			logging.error("Unable to read %s key from v2 data: %s" % (v2key, se))
+			exit(2)
+
+		is_json=True
+		try:
+			v2value = json.loads(so.split("\n")[0])
+		except ValueError:
+			v2value = so.split("\n")[0]
+			is_json=False
+
+		# check if they key is in v3 data
+		so, se, rc = runCommand("ETCDCTL_API=3 %s get %s -w json" % (ectl3, v2key))
+		# The rc or the se does not report if the key is missing or not.
+		# Either the v3value is empty or it is not
+		# TODO(jchaloup) catch the exception if the json.loads fails
+		etcddata = json.loads(so)
+		if not keyExists(etcddata):
+			logging.error("Key '%s' not found in v3 data" % v2key)
+			exit(3)
+
+		# v3 value from the json is base64 encoded
+		if is_json:
+			v3value = json.loads(base64.b64decode(etcddata["kvs"][0]["value"]))
```

Comment:
An easy way to provide the right type is to use ``six.b``. This will be a ``str`` on Python 2 and ``bytes`` on Python 3.

```python
v3value = json.loads(base64.b64decode(six.b(etcddata["kvs"][0]["value"])))
```